### ISSUE TITLE
OTP support?

### ISSUE THREAD

**Issue by dmorn** [FiloSottile/passage#47]
Is there any interest in adding OTP support in passage? I would be interested in knowing how you @FiloSottile suggest to deal with that, I used the pass-otp extension.

**Comment by remko** [comment]
There is [a PR for `pass-otp`](https://github.com/tadfisher/pass-otp/pull/178) that adds support for passage.

**Comment by ralphptorres** [comment]
If you're on Arch, there's also [pass-otp-age](https://aur.archlinux.org/packages/pass-otp-age) which applies remko's patch from the pr above. This package is transitionary; I'll remove it once the pr is merged.

**Comment by remko** [comment]
FYI, [the PR for `pass-otp`](https://github.com/tadfisher/pass-otp/pull/178) that adds support for passage has been merged.

**Comment by per-review** [comment]
I submitted [another PR](https://github.com/tadfisher/pass-otp/pull/212) with instructions for how to use pass-otp with passage:

> To use with [passage](https://github.com/FiloSottile/passage), set the variable `SYSTEM_EXTENSION_DIR` to one of passage’s extensions directories (by default `$HOME/.passage/extensions` and `/usr/lib/passage/extensions`) when installing [pass-otp]:
>
> ```
> sudo make install SYSTEM_EXTENSION_DIR=/usr/lib/passage/extensions
> ```

Edit: Fix variable position. It cannot precede the command.

**Comment by yushisora** [comment]
I have put the `OTP.bash` script in $PASSAGE_EXTENSIONS_DIR but how do I use the OTP extension?

**Comment by 53053** [comment]
> I have put the `OTP.bash` script in $PASSAGE_EXTENSIONS_DIR but how do I use the OTP extension?

for fish (translate to bash or your shell of choice, you need export these basically):

```fish
set -gx PASSWORD_STORE_ENABLE_EXTENSIONS true
set -gx PASSWORD_STORE_EXTENSIONS_DIR "$HOME/.passage/extensions" # or wherever you put it
```

now you should be able to use `pass otp example.com`
